infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutAlignmentDao.kt

```diff
@@ -203,80 +203,130 @@ class LayoutAlignmentDao(jdbcTemplateParam: NamedParameterJdbcTemplate?) : DaoBa
     }
 
     fun fetchSegmentGeometriesAndPlanMetadata(
-        alignmentId: IntId<LayoutAlignment>,
-        boundingBox: BoundingBox?
+        alignmentVersion: RowVersion<LayoutAlignment>,
+        metadataExternalId: Oid<*>?,
+        boundingBox: BoundingBox?,
     ): List<SegmentGeometryAndMetadata> {
+        //language=SQL
         val sql = """
-            with 
+            with
```

Comment:
SQL on tosiaan melko rankka (eikä alkuperäinenkään ollut ihan pieni). Sitä on koitettu pilkkoa ymmärrettäviin nimettyihin osiin tuolla with-syntaksilla mutta ei silti super luettava. Tuosta voisi tehdä myös viewin mutta käyttöpaikka on ainoastaan tämä joten päätin toistaiseksi jättää näin.

Tämä uusi extendattu versio toimii siis niin että se hakee raiteen kaikki metadataosiot, grouppaa ja yhdistelee ne jo kannassa ja lopuksi suodattaa niistä ne jotka osuu bounding boxin alueelle.

Reply:
Hyviä muutoksia 👏 Tämä on vaan tosiaan sen verran monimutkainen SQL, että sen voisi käydä vaikka ihan selittäen läpi tyyliin dailyssä. Tämän hanskaaminen kokonaisuudessaan pelkän reviewin perusteella menee nopeasti mun oman aivokapasiteetin yli 😄